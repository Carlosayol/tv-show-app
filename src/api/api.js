import axios from 'axios';
import { BASE_URL, API_KEY } from '../config/config';

export const getRequest = async (url) => {
  const API_URL = `${BASE_URL}${url}?api_key=${API_KEY}`;
  const response = await axios.get(API_URL)
  return response.data
};