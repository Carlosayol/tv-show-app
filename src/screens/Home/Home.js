import { Dimensions, View, Text, StyleSheet } from 'react-native'
import React from 'react'
import Constants from '../../utils/Constants'
import PopularSeries from '../../components/Popular'
import RecommendedSeries from '../../components/Recommended'

const deviceHeight = Dimensions.get('window').height;

const Home = () => {
  return (
    <View style={styles.container}>
      <PopularSeries />
      <RecommendedSeries />
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Constants.secondaryColor,
    height: deviceHeight,
  },
});

export default Home