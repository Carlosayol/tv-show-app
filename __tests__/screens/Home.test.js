import React from "react";
import renderer from "react-test-renderer";
import { render, cleanup, waitFor } from "@testing-library/react-native";

import Home from "../../src/screens/Home";
import { getRequest } from "../../src/api/api";

const mockedNavigate = jest.fn();
jest.mock("@react-navigation/native", () => ({ useNavigation: () => ({ navigate: mockedNavigate }) }));
jest.mock("../../src/api/api");

const createTestProps = (props) => ({
  navigation: {
    navigate: jest.fn(),
  },
  ...props,
});

describe("Home Screen", () => {
  let props;
  beforeEach(() => {
    props = createTestProps({});
    getRequest.mockResolvedValue({
      results: [],
    });
  });
  afterEach(cleanup);

  it("render base element", () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree.children.length).toBe(2);
  });

  it("renders correctly", () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("find Popular section", async () => {
    const { getByText } = render(<Home {...props} />);
    await waitFor(() => {
      expect(getByText("Popular")).toBeDefined();
    });
  });
});
