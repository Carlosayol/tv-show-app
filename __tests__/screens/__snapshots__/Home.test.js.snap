// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#191919",
      "flex": 1,
      "height": 1334,
    }
  }
>
  <View>
    <ActivityIndicator
      color="#FFFFFF"
      size="large"
    />
  </View>
  <View>
    <ActivityIndicator
      color="#FFFFFF"
      size="large"
    />
  </View>
</View>
`;
